specs/specCircuitBreakerBooster.ts

```diff
@@ -1,23 +1,32 @@
 import { CircuitBreakerPage, NameServiceState, GreetingResult } from '../pages';
 import { browser, protractor } from 'protractor';
 
-// The jasmine typings are brought in via DefinitelyTyped ambient typings.
 describe('Circuit breaker booster page', () => {
 
+  let EC = protractor.ExpectedConditions;
+
+  beforeEach(async () => {
+    let circuitBreakerPage = new CircuitBreakerPage();
+    await circuitBreakerPage.get();
+    let text = await circuitBreakerPage.getNameServiceStateElement().getText();
+    if (text === NameServiceState.Failure) {
+      await circuitBreakerPage.clickToggle();
+      browser.wait(EC.textToBePresentInElement(circuitBreakerPage.getNameServiceStateElement(), NameServiceState.Working), 1000);
+    }
+  });
+
   it('Click invoke without toggle', () => {
     let circuitBreakerPage = new CircuitBreakerPage();
     circuitBreakerPage.get();
-    let EC = protractor.ExpectedConditions;
     browser.wait(EC.textToBePresentInElement(circuitBreakerPage.getNameServiceStateElement(), NameServiceState.Working), 1000);
+    browser.wait(EC.textToBePresentInElement(circuitBreakerPage.getGreetingElement(), ""), 1000);
```

Comment:
Would it be enough to wait for the elemente to be present? Waiting for empty string to be present inside the element looks strange :-)